modules/core/src/bitgo.ts

```diff
@@ -141,171 +149,437 @@ const createResponseErrorString = function(res) {
+    // functions that use it.
+    this._validate = params.validate === undefined ? true : params.validate;
+
+    // Create superagent methods specific to this BitGo instance.
+    const methods = ['get', 'post', 'put', 'del', 'patch'];
+
+    if (!params.proxy && process.env.BITGO_USE_PROXY) {
+      params.proxy = process.env.BITGO_USE_PROXY;
+    }
 
-  this._microservicesUrl = params.microservicesUri;
-  this._baseApiUrl = this._baseUrl + '/api/v1';
-  this._baseApiUrlV2 = this._baseUrl + '/api/v2';
-  this._user = null;
-  this._keychains = null;
-  this._wallets = null;
-  this._clientId = params.clientId;
-  this._clientSecret = params.clientSecret;
-  this._token = params.accessToken || null;
-  this._refreshToken = params.refreshToken || null;
-  this._userAgent = params.userAgent || 'BitGoJS/' + this.version();
-  this._promise = Bluebird;
+    if ((process as any).browser && params.proxy) {
+      throw new Error('cannot use https proxy params while in browser');
+    }
 
-  // whether to perform extra client-side validation for some things, such as
-  // address validation or signature validation. defaults to true, but can be
-  // turned off by setting to false. can also be overridden individually in the
-  // functions that use it.
-  this._validate = params.validate === undefined ? true : params.validate;
+    this._proxy = params.proxy;
 
-  // Create superagent methods specific to this BitGo instance.
-  this.request = {};
-  const methods = ['get', 'post', 'put', 'del', 'patch'];
+    for (const index in methods) {
+      const method = methods[index];
+      this[method] = this.createPatch(method);
+    }
 
-  if (!params.proxy && process.env.BITGO_USE_PROXY) {
-    params.proxy = process.env.BITGO_USE_PROXY;
-  }
+    // capture outer stack so we have useful debug information if fetch constants fails
+    const e = new Error();
 
-  if ((process as any).browser && params.proxy) {
-    throw new Error('cannot use https proxy params while in browser');
+    // Kick off first load of constants
+    this.fetchConstants({}, function(err) {
+      if (err) {
+        // make sure an error does not terminate the entire script
+        console.error('failed to fetch initial client constants from BitGo');
+        debug(e.stack);
+      }
+    });
   }
 
-  this.proxy = params.proxy;
-
-  const self = this;
-
-  // This is a patching function which can apply our authorization
-  // headers to any outbound request.
-  const createPatch = function(method) {
-    return function() {
-      let req = superagent[method].apply(null, arguments);
-      if (self.proxy) {
-        req = req.proxy(self.proxy);
+  /**
+   * This is a patching function which can apply our authorization
+   * headers to any outbound request.
+   * @param method
+   */
+  private createPatch(method: string): (url: string, callback?: NodeCallback<superagent.Response>) => superagent.Request {
+    const self = this;
+    return function(...args) {
```

Comment:
is ```args``` correct here, or should it be ```arguments``` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments

Reply:
oh nvm i think thats answered here https://stackoverflow.com/questions/42184674/what-is-the-meaning-of-args-three-dots-in-a-function-definition

Reply:
They are equivalent, but ...args plays better with typescript (`arguments` is a very strange beast)